docs/tutorials/acme/rfc2136-provider.rst

```diff
@@ -0,0 +1,171 @@
+``dnssec-keygen``. This command-line tool generates a named private key
+that is used for signing TSIG requests. When a request is signed, both
+the signature and the name of the private key are attached to the
+request in an unencrypted form. In this manner, when the request is
+received, the name of the private key can be used to by the recipient to
+find the private key itself, build a new signature with it, and compare
+the two for acceptance.
+
+Since there are dozens of ways to have your ``named`` server
+misconfigured, we’ll use ``nsupdate`` to test that the server behaves as
+expected before we get there.
+https://debian-administration.org/article/591/Using_the_dynamic_DNS_editor_nsupdate
+is a solid breakdown of how to use the tool.
+
+To get started, we’ll simply run ``nsupdate -k <keyID>`` where keyID is
+the value returned from ``dnssec-keygen``. This will read the key from
+disk and provide a command prompt to issue commands. In general, we want
+to write a simple TXT RR and make sure we can delete it.
+
+::
+
+   $ nsupdate -k <keyID>
+   > update add www1.example.com txt testing
+   > send
+   > … test here with ``nslookup``
+   > update delete www1.example.com txt
+   > send
+   > … test here with ``nslookup``
+
+Any failures to write, read or delete the record will mean that
+cert-manager will not be able to do so either, no matter how well it is
+configured.
+
+Configuration Step 2 - Set up cert-manager
+------------------------------------------
+
+Now we get to the fun stuff, seeing everything work. Remember that we
+need to set up the ACME DNS-01 issuer and challenge mechanism as well as
+the ``rfc2136`` provider. Since the documentation covers the other parts
+sufficiently, let’s focus on the provider here.
+
+.. code:: yaml
+       rfc2136:
+         nameserver: <address of authoritative nameserver configured above>
+         tsigKeyName: <key name used in `dnssec-keygen`, use something semantically meaningful in both environments>
+         tsigAlgorithm: HMACSHA512 // should be matched to the algo you chose in `dnssec-keygen`
+         tsigSecretSecretRef:
+           name: <the name of the k8s secret holding the TSIG key.. not the key itself!>
+           namespace: <define if not `default`>
+           key: <name of the key *inside* the secret>
+
+Example:
+
+.. code:: yaml
+
+       rfc2136:
+         nameserver: 1.2.3.4:53
+         tsigKeyName: example-com-secret
+         tsigAlgorithm: HMACSHA512
+         tsigSecretSecretRef:
+           name: tsig-secret
+           namespace: cert-manager
+           key: tsig-secret-key
+
+For this example configuration, we’ll need the following two commands.
+The first, on your ``named`` server generates the key. Note how
+``example-com-secret`` is both in the ``tsigKeyName`` above and the
+``dnssec-keygen`` command that follows.
+
+::
+
+   dnssec-keygen -r /dev/urandom -a HMAC-SHA512 -b 512 -n HOST example-com-secret
```

Comment:
Is this an example of the command that needs to be run as part of step (1) (`Set up your DNS server for secure dynamic updates`)?

I'm slightly confused about the distinction between the two, and the YAML example provided at the start of step (2) also references `<key name used in `dnssec-keygen`, use something semantically meaningful in both environments>`

Reply:
Thanks! The feedback is critical and I was hoping to have the peer review and edits.

This is indeed necessary for setting up the DNS server. The key that is generated here needs to be provided to the DNS server as a shared secret so the signatures can be validated.